grpclib/config.py

```diff
@@ -55,7 +55,7 @@ def _with_defaults(
 ) -> _ConfigurationType:
     defaults = {}
     for f in fields(cls):
-        if f.default is _DEFAULT:
+        if getattr(cls, f.name) is _DEFAULT:
```

Comment:
Before this change, if the parameter had a default value for the server or for the client, it will not matter which one we set, since it will be set to the default value for the server and client. This test shows this behavior test_change_default